Make promotion messaging work for SW. Fix minor CSS bug

srallen opened the pull request:
Staging branch URL: https://sw-promotion-message.pfe-preview.zooniverse.org/projects/zooniverse/gravity-spy?env=production

Fixes #4167. We can't quite test this for Snapshot Wisconsin Challenge yet, so the staging link is to test that this doesn't break Gravity Spy. This can be tested by going to a previous level. The promotion dialog will pop up when the subject queue is empty. I did enough classifications to get the zootester1 account promoted to level 2 on gravity spy in case you don't have an account that's been promoted yet. If you go to level 1, do enough classifications to clear the subject queue, you should see Gravity Spy's promotion dialog with their specific text. 

The other change was to make it so that the custom sign in prompt will only show for Gravity Spy. Since SW:C will be done with focus groups, they'll make sure the users are signed in. 

I also fixed some minor linter errors and a CSS bug with the mini-course.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

eatyourgreens commented on app/pages/project/classify.cjsx:
```diff
@@ -305,9 +305,25 @@ module.exports = React.createClass
 
     @loadAppropriateClassification(@props) if @props.workflow?
 
+  getPromotionMessage: (props) ->
```
Could this be set in [the translations wrapper for projects](/zooniverse/Panoptes-Front-End/blob/master/app/pages/project/project-translations.jsx), then referenced as `props.translation.promotion_message`:

srallen replied:
I've moved the default promotion message into the project translation wrapper. I'm hesitant to put the custom messages for Gravity Spy and Snapshot Wisconsin in there because these being hard coded is temporary until it gets moved into the project contents in Panoptes. Once those are there, we can check for the presence of those project content strings or fallback to the default messaging. Does this work?

eatyourgreens replied:
The translation wrapper should take care of that fallback for you, shouldn't it?

I don't think it makes a big difference here, but I'd prefer if components expect their language strings to be provided externally, via the `translation` prop, rather than defining them themselves.

srallen replied:
Ok, I can do another quick refactor to have the handling of this in the translation wrapper.

srallen commented:
@eatyourgreens I refactored to actually determine which message to use in the `WorkflowAssignmentDialog` and the default message is in the en.js file. I also added the button strings to the en.js file. Once the promotion message is added to the project contents on the API, then it can be passed down properly through the project translations wrapper. 

I dismissed the previous approval since this has changed enough to warrant another full review. This can be tested a couple of ways:

For gravity spy, use an account that has been promoted at least to level two: https://sw-promotion-message.pfe-preview.zooniverse.org/projects/zooniverse/gravity-spy?env=production

The zootester1 account has been so you can use that if your own account hasn't been promoted yet on the GS project. If you do classifications on level 1, you should see the dialog with the correct message once the subject queue is clear.

On staging, you can use my test project to see the default message: https://sw-promotion-message.pfe-preview.zooniverse.org/projects/srallen086/workflow-assignment-testing

You'll have to promote your own account to test this. As an admin, update your preferences's settings property, `settings.workflow_id`,  set to `'2334'` and classify on the beginner level. Once the queue is empty, you'll see the dialog.

eatyourgreens commented on app/locales/en.js:
```diff
@@ -9,7 +9,12 @@ export default {
     talk: 'Talk',
     tutorialButton: 'Show the project tutorial',
-    miniCourseButton: 'Restart the project mini-course'
+    miniCourseButton: 'Restart the project mini-course',
```
These should be added to the other language files, so that people know they're available to be translated.

eatyourgreens commented:
Looks good to me. The new language strings should be added to the other language files at some point, but I don't think that should hold up merging this.
